echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
-              haul_data
-              .groupby( [ 'stratum_num' ] )
-              .apply( lambda df: pd.Series( { 'stratum_weight': df.haul_weight.sum( ) } ) )
-              .reset_index( )
-              .assign( station = 1 ) ,
-              # Specimen data
-              specimen_data 
-              .groupby( [ 'stratum_num' ] )
-              .apply( lambda df: pd.Series( { 'stratum_weight': df.weight.sum( ) } ) )
-              .reset_index()
-              .assign( station = 2 )
-         ]
+    ### Process biological datasets
+    # ---- Process haul data ( Station 2 - unsexed - unaged )
+    haul_strata_weight = (
+        haul_data
+        .groupby( 'stratum_num' )[ 'haul_weight' ]
+        .sum( )
+        .to_frame( 'stratum_weight' )
+        .reset_index( )
+        .assign( group = 'unaged' )
+    ) 
+
+    # ---- Process specimen data ( Station 1 - sexed - aged )
+    specimen_strata_weight = (
+        specimen_data
+        .groupby( 'stratum_num' )[ 'weight' ]
+        .sum( )
+        .to_frame( 'stratum_weight' )
+        .reset_index( )
+        .assign( group = 'aged' )
     )
 
+    ### Merge the two dataframes
+    # Yields the summed weights per stratum and station (or aged/unaged designation)
+    weight_strata_aged_unaged = pd.concat( [ haul_strata_weight , specimen_strata_weight ] )
+ 
     ### Sum weights for each stratum
     weight_strata = (
-            weight_strata_station
-            .groupby( [ 'stratum_num' ] )
-            .apply( lambda x: pd.Series( { 'weight_stratum_total': x.stratum_weight.sum( ) } ) )            
-            .reset_index( )
-        )
+        weight_strata_aged_unaged
+        .groupby( "stratum_num" )[ "stratum_weight" ]
+        .sum( )
+        .to_frame( 'weight_stratum_all' )
+        .reset_index( )
+    )
     
     ### Carriage return
-    return weight_strata_station , weight_strata
+    return weight_strata_aged_unaged , weight_strata
 
-def sum_strata_length_age_sex_weight( haul_data: pd.DataFrame ,
-                                      specimen_data: pd.DataFrame ,
-                                      length_intervals: np.ndarray ,
-                                      age_intervals: np.ndarray ):
-    
-    ### Sum weights from each station and strata 
-    weight_strata_station , weight_strata = sum_strata_weight( haul_data ,
-                                                               specimen_data )
+def compute_aged_unaged_proportions( specimen_data: pd.DataFrame ,
+                                     weight_strata: pd.DataFrame ):
+    """
+   Compute the overall weight proportions for aged and uanged fish within each stratum 
+
+    Parameters
+    ----------
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    weight_strata: pd.DataFrame
+        Dataframe contained summed weights of both aged and unaged fish
+    """   
+    ### Calculate adult:all age ratio for each stratum
+    # ---- Drop unaged fish
```

Comment:
thought might be a good idea to be specific from which station of data these unaged fish are dropped
```suggestion
    # ---- Drop unaged fish from station 1 data (where most fish are aged)
```